### PR TITLE
`Development`: Fix management dashboard queries for PostgreSQL applications

### PR DESCRIPTION
**Pull request by julian-christl**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [X] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Server
- [X] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [X] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [X] I documented the Java code using JavaDoc style.

### Motivation and Context

PostgreSQL apparently has issues comparing supplied parameters since the Spring Boot 3 upgrade with `NULL` which caused 15 tests to fail.

### Description

I separated the four-parted query into four different queries. Semantically, it is the very same. There are no additional database calls.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Admin
- PostgreSQL Testserver
- Existing populated courses

Open the management view as an instructor and ensure you can see all courses you should see in the expected manner. Do the same as an admin.

### Code Review

Make sure that the functionality is indeed semantically the same!

#### Performance Review
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| CourseService.java | 92.4% |                            |

## Summary by CodeRabbit

- **Refactor**
	- Improved the efficiency of fetching courses for management overview.
- **New Features**
	- Added functionality to filter courses based on management group names and their active/end status, enhancing course management capabilities.

### CODE REVIEW

**Review by mateusmm01 — APPROVED** [review]
code lgtm!

**Review by Strohgelaender — DISMISSED** [review]
lgtm

**Review by mateusmm01 — DISMISSED** [review]
Tested on ts5 as inctructor and admin. Works as expected

**Review by basak-akan — APPROVED** [review]
Tested on ts6, management view works as expected as instructor and admin

File: src/main/java/de/tum/in/www1/artemis/repository/CourseRepository.java [hunk 1]
```diff
@@ -283,28 +283,54 @@ GROUP BY SUBSTRING(CAST(s.submissionDate AS string), 1, 10), p.student.login
...
-     * @param userGroups the user groups of the user to fetch the courses for (ignored if the user is an admin)
-     * @return a list of courses for the overview
+     * @param now the current time
+     * @return a list of courses that are not ended yet
+     */
+    @Query("""
+            SELECT c
+            FROM Course c
+            WHERE c.endDate IS NULL
+                OR c.endDate >= CAST(:now AS timestamp)
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/repository/CourseRepository.java:295]: Is `CAST(:now AS timestamp)` really needed here? We have quite a few places where it does not seem necessary and where we simply use `:now` which is much simpler and easier to understand

Comment (julian-christl) [reply src/main/java/de/tum/in/www1/artemis/repository/CourseRepository.java]: Indeed unnecessary. We already use it without throughout the system so we know that it works without. I removed the cast.

**Pull request merged**